tests/model/test_scan_data.py

```diff
@@ -108,8 +108,8 @@ def test_swap():
     scan1.swap(scan2)
     assert scan2.get_image_range() == (1, 20)
     assert scan1.get_image_range() == (40, 60)
-    assert scan2.get_oscillation() == (0.0, 1.0)
-    assert scan1.get_oscillation() == (10.0, 2.0)
+    assert scan2.get_oscillation() == pytest.approx((0.0, 1.0))
```

Comment:
Why now approx? For record. ISTR, all these values should be exactly representable as `float` or `double`?

Reply:
Reason for approx described in corresponding DIALS PR [#2408](https://github.com/dials/dials/pull/2408)

Reply:
It depends if we're asserting if the output values are _precisely_ the unchanged input values, though even with conversion these specific instances shouldn't change.

I'd lean towards no-approx, given that this is testing `swap` and if `swap` changes the values at all, that's probably n bug, but in that case unrepresentable values should probably be tested anyway.

This test actually had values change without the `approx` here?

Reply:
Okay so the actual answer is that these are converted to radians, then stored as calculated absolute values `(S, W) -> [S, S+W, S+2W, ....]`, and then on asking for `get_oscillation` again converted back to `[S, S+W, ...] -> (S, S+W-S)`. So I'm not surprised that these have differences.

To do what this test is actually trying to do, we should probably read out `scan1.get_oscillation()` before swapping, and compare to that.